prdoc/pr_5130.prdoc

```diff
@@ -0,0 +1,31 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Add SingleAssetExchangeAdapter
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      SingleAssetExchangeAdapter is an adapter in xcm-builder that can be used
+      to configure the AssetExchanger in XCM to use pallet-asset-conversion.
+      It can be configured as follows:
+      ```rust
+      pub type AssetExchanger = SingleAssetExchangeAdapter<
+        // pallet-assets-conversion, as named in `construct_runtime`.
+        AssetConversion,
+        // The fungibles implementation that brings together all assets in pools.
+        // This may be created using `fungible::UnionOf` to mix the native token
+        // with more tokens.
+        Fungibles,
+        // The matcher for making sure which assets should be handled by this exchanger.
+        Matcher,
+      >;
+      ```
+      It's called "single asset" since it will only allow exchanging one asset for another.
+      It will error out if more than one asset tries to be exchanged.
+
+      Also, a new method was added to the `xcm_executor::traits::AssetExchange` trait:
+      `quote_exchange_price`. This is used to get the exchange price between two assets.
+      This is a breaking change. If you have implemented the trait, you must add the new method.
```

Comment:
Like you suggested, this would not be the case with a default `None` implementation.

```suggestion
```